### PR TITLE
[hotfix][table][tests] Improve code - use assertJ

### PR DESCRIPTION
**Pull request by matriv**
## What is the purpose of the change

Small code improvements around test in `flink-table` module

## Brief change log

  - Since `table.exec.legacy-cast-behaviour` is disabled by default, there is no need to explicitely set it in the configuration of the Cast related IT tests.
  - Call `super.getConfiguration()` to inherit the configuration set in the superclass and add/override with more config options in the subclass.
  - Fix warnings
  - Use assertJ
  - Remove unused code
  - Replace deprecated ExpectedException with assertJ's `assertThatThrownBy` and improve certain assertions by asserting the Exception class or the Exception message where missing.

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): **no**
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: **no**
  - The serializers: **no**
  - The runtime per-record code paths (performance sensitive): **no**
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: **no**
  - The S3 file system connector: **no**

## Documentation

  - Does this pull request introduce a new feature? **no**
  - If yes, how is the feature documented? **not applicable**

### CODE REVIEW

File: flink-table/flink-table-runtime/src/test/java/org/apache/flink/table/data/DataStructureConvertersTest.java [hunk 1]
```diff
@@ -367,36 +365,40 @@ public static List<TestSpec> testData() {
...
-
-                final DataStructureConverter<Object, Object> toConverter =
-                        simulateSerialization(DataStructureConverters.getConverter(toDataType));
-                toConverter.open(DataStructureConvertersTest.class.getClassLoader());
-
-                assertThat(toConverter.toExternalOrNull(internalValue)).isEqualTo(to.getValue());
+            if (testSpec.expectedErrorMessage != null) {
+                assertThatThrownBy(
+                                () ->
+                                        simulateSerialization(
```
Comment (twalthr) [inline flink-table/flink-table-runtime/src/test/java/org/apache/flink/table/data/DataStructureConvertersTest.java:376]: remove in the `simulateSerialization` in this case

File: flink-table/flink-table-runtime/src/test/java/org/apache/flink/table/runtime/typeutils/RowDataSerializerTest.java [hunk 2]
```diff
@@ -347,26 +345,24 @@ public void testCopy() {
...
+                                            row,
+                                            serializer.copy(
+                                                    row, new GenericRowData(row.getArity() + 1)),
+                                            false))
+                    .isInstanceOf(IllegalArgumentException.class);
         }
     }
 
     /** Class used for concurrent testing with KryoSerializer. */
-    private static class WrappedString {
```
Comment (twalthr) [inline flink-table/flink-table-runtime/src/test/java/org/apache/flink/table/runtime/typeutils/RowDataSerializerTest.java:364]: I'm sure this had a deeper meaning. Let's keep it.

**Review by twalthr — APPROVED** [review]
Thanks @matriv. LGTM I had two minor comments that I will fix while merging.

**Comment by matriv** [comment]
thank you @twalthr!

**Pull request closed without merging**